app/controllers/api/v0/events_controller.rb

```diff
@@ -6,11 +6,11 @@ def create
     render(json: error, status: error.status_code) and return if error
 
     inbound_binding.events.each do |event|
-      # first, schema validate the event.data object
+      event.data[:uuid] = CompactUuid.pack(current_user_uuid) if current_user_uuid
 
-      event.data[:user_uuid] = current_user_uuid if current_user_uuid
+      avro_encoded_data = KafkaAvroTurf.instance.encode(event.data, schema_name: event.type)
 
-      KafkaClient.produce(data: event.data, topic: event.topic)
+      KafkaClient.produce(data: avro_encoded_data, topic: event.topic)
```

Comment:
Can we do the [asynchronous production approach](https://github.com/a15k/interactions-api/blob/master/lib/kafka_helpers.rb#L24) to limit load on Kafka? (batch production)... actually I see some of this code is already in the baseline, just not used yet.

Reply:
@jpslav we are already doing async calls to kafka thru this code. here the KafkaClient uses a async producer.  i've tested this, it works.  it's configured thru these secrets configuration:
```
    delivery_threshold: 5
    delivery_interval: 5
```